pyanaconda/modules/storage/partitioning/automatic/automatic_partitioning.py

```diff
@@ -122,37 +123,53 @@ def _get_luks_format_args(storage, request):
         }
 
     @staticmethod
-    def _get_partitioning(storage, scheme, excluded_mount_points=()):
+    def _get_partitioning(storage, scheme, request: PartitioningRequest):
         """Get the partitioning requests for autopart.
 
         :param storage: blivet.Blivet instance
         :param scheme: a type of the partitioning scheme
-        :param excluded_mount_points: a list of mount points to exclude
+        :param request: partitioning parameters
         :return: a list of full partitioning specs
         """
-        requests = []
+        specs = []
+        swap = None
 
-        for request in get_default_partitioning():
+        # Create partitioning specs based on the default configuration.
+        for spec in get_default_partitioning():
             # Skip mount points excluded from the chosen scheme.
-            if request.schemes and scheme not in request.schemes:
+            if spec.schemes and scheme not in spec.schemes:
                 continue
 
             # Skip excluded mount points.
-            if (request.mountpoint or request.fstype) in excluded_mount_points:
+            if (spec.mountpoint or spec.fstype) in request.excluded_mount_points:
                 continue
 
+            # Detect swap.
+            if spec.fstype == "swap":
+                swap = spec
```

Comment:
Is it a given there will be only one swap?

Reply:
I can't imagine a scenario where autopart would create more than one swap. But maybe it would be worth it to future/fool proof this by changing this.

Reply:
The code should be fine. The `swap` variable is basically just a flag at this moment. We just need to add one of there is none. The swap size is then recalculated for all of them.